cudamapper/src/index_cache.cu

```diff
@@ -0,0 +1,244 @@
+/*
+* Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+*
+* NVIDIA CORPORATION and its licensors retain all intellectual property
+* and proprietary rights in and to this software, related documentation
+* and any modifications thereto.  Any use, reproduction, disclosure or
+* distribution of this software and related documentation without an express
+* license agreement from NVIDIA CORPORATION is strictly prohibited.
+*/
+
+#include "index_cache.cuh"
+
+#include "index_host_copy.cu"
+
+#include <claragenomics/cudamapper/index.hpp>
+#include <claragenomics/io/fasta_parser.hpp>
+
+namespace claragenomics
+{
+namespace cudamapper
+{
+
+IndexDescriptor::IndexDescriptor(read_id_t first_read, read_id_t number_of_reads)
+    : first_read_(first_read)
+    , number_of_reads_(number_of_reads)
+    , hash_(0)
+{
+    generate_hash();
+}
+
+read_id_t IndexDescriptor::first_read() const
+{
+    return first_read_;
+}
+
+read_id_t IndexDescriptor::number_of_reads() const
+{
+    return number_of_reads_;
+}
+
+std::size_t IndexDescriptor::get_hash() const
+{
+    return hash_;
+}
+
+void IndexDescriptor::generate_hash()
+{
+    static_assert(sizeof(size_t) == 8, "only 64-bit values supported, adjust element_mask and shift_bits");
+
+    // populate lower half of hash with one value, upper half with the other
+    std::size_t element_mask = 0xFFFFFFFF; // for 64 bits
+    std::uint32_t shift_bits = 32;         // for 64 bits
```

Comment:
Optional: Any reason you don't mark these `constexpr` or at least `const`?

Reply:
No, I just forgot to add it once I removed the `if` statement.
Done